core/conn/unixodbc/odbc/odbcclient/unixcli/cli/sqltocconv.cpp

```diff
@@ -1555,6 +1555,7 @@ unsigned long ODBC::ConvertSQLToC(SQLINTEGER	ODBCAppVersion,
 								return IDS_22_003;
 							}
 							srcLength = TransStringLength + 1;
+                            DataLen = sizeof(DATE_STRUCT);
```

Comment:
I don't understand why this code change is correct. It looks like we are converting between a variety of C numeric types and ODBC varchar types. Why would DATE enter into it?